File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        // int n = digits.length;
        // //int[] arr = new int[n+1];
        
        // for(int i=n-1;i>=0;i--){
        //     if(digits[n-1]!=9){
        //         digits[i]++;
        //         break;
        //     }else{
        //         digits[0]=0;
        //     }
        // }
        // if(digits[0]==0){
        //     int[] arr = new int[n+1];
        //     arr[0]=1;
        //     return arr;
        // }


        // return digits;
          int n=digits.length;
        for(int i=n-1;i>=0;i--)
        {
            if(digits[i] != 9)
            {
                digits[i]++;
                break;
            }
            else{
                digits[i]=0;
            }
        }
        if(digits[0] == 0)
        {
            int[] res = new int[n+1];
            res[0]=1;
            return res;
        }

        return digits;   
    }
}

File: 66-plus-one/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
     for (let i = digits.length - 1; i >= 0; i--) {
        if (digits[i] != 9) {
            digits[i]++;
            return digits;
        } else {
            digits[i] = 0;
        }
    }
    
    // If all the digits were 9, we need an array with one more digit, starting with 1 followed by all 0s
    let arr = new Array(digits.length + 1).fill(0);
    arr[0] = 1;
    return arr;
};